fix: remove `merge-deep` from the validation

Megapixel99 opened the pull request:
We used to use `merge-deep` to assign parameters to a new object; however, we can use `structuredClone`  to do the same thing and remove the dependency on `merge-deep`. `structuredClone` only works on node 17+ but node 16 reached EOL in September 2023 per https://endoflife.date/nodejs and https://nodejs.org/en/blog/announcements/nodejs16-eol.

Megapixel99 commented:
Ran the tests locally with all of the same versions as above (and 23), all of the tests passed (except for node 16).

Versions used to test:
- v23.0.0 (passed)
- v22.10.0 (passed)
- v20.18.0 (passed)
- v18.20.4 (passed)
- v16.20.2 (failed)

enov reviewed (approved):
Great PR! `merge-deep` is unnecessary nowadays and has some require magic conflicting with some js bundlers. I pulled the branch and tests against Node.js v18, v20, and v22 were successful.

Megapixel99 commented:
The tests are failing due to a newer version of `path-to-regexp` being used. I will install the newer version and update the tests

Megapixel99 commented:
> The tests are failing due to a newer version of `path-to-regexp` being used. I will install the newer version and update the tests

I updated all of the dependencies and updated the tests, I can make the changes part of this PR or make a new PR with the changes to the tests. Aside from the changes to `package.json` and adding onto the tests (just to make sure I did not break anything) only two lines were changed to ensure the tests pass.

wesleytodd commented on lib/validate.js:
```diff
@@ -37,7 +36,7 @@ module.exports = function makeValidatorMiddleware (middleware, schema, opts) {
 
   function makeValidator () {
-    const reqSchema = merge({}, BASE_REQ_SCHEMA)
+    const reqSchema = structuredClone(BASE_REQ_SCHEMA)
```
I guess I just wonder if a more simple approach to this would be enough? I am not opposed to this, but since we are not doing this for the specific reason `structuredClone` is meant to serve I wonder if there are any unexpected consequences of this approach.

wesleytodd reviewed (approved):
Lets go with this for now and we can decide after the work to support v5 if it is good to keep.

wesleytodd commented:
The tests are broken on `main` because `express@4.20.0` broke our regex handling. Going to merge this despite the tests not passing because I think it is a good change and then fix the remaining issues after.
